internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -428,25 +428,34 @@ func (r *PreprovisioningImageReconciler) getIronicAgentImageFromUserOverride(log
 	return ""
 }
 
-func (r *PreprovisioningImageReconciler) getIronicConfigFromBMOConfig(ctx context.Context, log logrus.FieldLogger, infraEnvInternal *common.InfraEnv) *ICCConfig {
+func (r *PreprovisioningImageReconciler) getIronicConfigFromBMOConfig(ctx context.Context, log logrus.FieldLogger) *ICCConfig {
 	iccConfig, err := r.BMOUtils.getICCConfig(ctx)
 	if err != nil {
 		log.WithError(err).Info("ICC configuration is not available")
 		return nil
 	}
 
-	var architectures []string
-	architectures, err = r.OcRelease.GetImageArchitecture(log, iccConfig.IronicAgentImage, infraEnvInternal.PullSecret)
+	log.Infof("Using ironic URLs from ICC config (Base: %s, Inspector: %s)", iccConfig.IronicBaseURL, iccConfig.IronicInspectorBaseUrl)
+	return iccConfig
+}
```

Comment:
This function is not that interesting now its just wrapping another single function call .. Can we just move `r.BMOUtils.getICCConfig` to the caller instead and get rid of the bonus wrapper function?